### PR TITLE
Integration test mechanism rework

### PR DESCRIPTION
**Pull request by dnavre**
Originally started as a rework of the integration test mechanism this expanded into something gargantuan.

This pull request includes the following tasks/topics:

- [x] Upgraded to Spring boot 2
- [x] Refactoring of API facade structure
- [x] Refactoring of API REST endpoint structure
- [x] API module package structure refactoring
- [x] Refactoring of persistence layer initialization procedures
- [x] Migration from XML based context initialization to @Configuration classes
- [x] API & worker modules are starting up
- [x] Refactoring of the API layer
- [x] Fix the worker not connecting to the rabbitMQ
- [x] Go over the dependency diagram, ensure that there are no conflicts/mistakes
- [x] API & worker modules  should be fully operational
- [x] Upgrade of Twillio sdk to version 7.38.1
- [ ] Test Kafka setup

### CODE REVIEW

File: report/pom.xml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <packaging>pom</packaging>
+    <artifactId>report</artifactId>
+
+
+    <build>
+        <plugins>
+            <plugin>
```
Comment (mikron) [inline report/pom.xml]:
```suggestion
            <plugin>
                <groupId>org.sonatype.plugins</groupId>
                <artifactId>nexus-staging-maven-plugin</artifactId>
                <configuration>
                    <skipNexusStagingDeployMojo>true</skipNexusStagingDeployMojo>
                </configuration>
            </plugin>
            <plugin>
```

Comment (dnavre) [reply report/pom.xml]: done

File: pom.xml [hunk 2]
```diff
@@ -139,21 +125,25 @@
...
         <!--Sonar config-->
         <sonar.core.codeCoveragePlugin>jacoco</sonar.core.codeCoveragePlugin>
         <sonar.language>java</sonar.language>
+        <sonar.dynamicAnalysis>reuseReports</sonar.dynamicAnalysis>
         <sonar.jacoco.includes>com.sflpro.*</sonar.jacoco.includes>
-        <sonar.jacoco.version>0.8.3</sonar.jacoco.version>
-        <jacoco.plugin.version>0.8.3</jacoco.plugin.version>
-        <google.guave.version>18.0</google.guave.version>
-        <apache.tika.version>1.20</apache.tika.version>
+        <sonar.jacoco.version>0.8.4</sonar.jacoco.version>
```
Comment (mikron) [inline pom.xml]:
```suggestion
        <sonar.jacoco.version>1.0.1</sonar.jacoco.version>
```

Comment (dnavre) [reply pom.xml]: Done. Let's see what happens :)

**Pull request merged**